benchmark/ReadMultipleDocumentsBench.php

```diff
@@ -0,0 +1,168 @@
+            'codec' => null,
+            'typeMap' => [
+                'array' => BSONArray::class,
+                'document' => BSONDocument::class,
+                'root' => BSONDocument::class,
+            ],
+            'accessor' => 'object',
+        ];
+
+        yield 'Raw BSON' => [
+            'codec' => null,
+            'typeMap' => ['root' => 'bson'],
+            'accessor' => 'bson',
+        ];
+    }
+
+    #[ParamProviders('provideParams')]
+    public function benchCursorToArray(array $params): void
+    {
+        $options = array_intersect_key($params, ['codec' => true, 'typeMap' => true]);
+
+        $collection = self::getCollection();
+
+        $collection->find([], $options)->toArray();
+    }
+
+    #[ParamProviders('provideParams')]
+    public function benchAccessId(array $params): void
+    {
+        $options = array_intersect_key($params, ['codec' => true, 'typeMap' => true]);
+
+        $collection = self::getCollection();
+
+        // Exhaust cursor and access identifier on each document
+        foreach ($collection->find([], $options) as $document) {
+            $this->accessId($document, $params['accessor']);
+        }
+    }
+
+    #[ParamProviders('provideParams')]
+    public function benchAccessNestedItem(array $params): void
+    {
+        $options = array_intersect_key($params, ['codec' => true, 'typeMap' => true]);
+
+        $collection = self::getCollection();
+
+        // Exhaust cursor and access identifier on each document
+        foreach ($collection->find([], $options) as $document) {
+            $this->accessNestedItem($document, $params['accessor']);
+        }
+    }
+
+    /** @param array|object $document */
+    private function accessId($document, string $accessor): void
+    {
+        switch ($accessor) {
+            case 'array':
+                assert(is_array($document));
+                $document['_id'];
+                break;
+
+            case 'bson':
+                assert($document instanceof Document);
+                $document->get('_id');
+                break;
+
+            case 'object':
+                assert(is_object($document));
+                $document->_id;
+                break;
+
+            default:
+                throw new InvalidArgumentException(sprintf('Invalid accessor "%s"', $accessor));
+        }
+    }
+
+    /** @param array|object $document */
+    private function accessNestedItem($document, string $accessor): void
+    {
+        switch ($accessor) {
+            case 'array':
+                assert(is_array($document));
+                $document['entities']['user_mentions'][0]['screen_name'];
+                break;
+
+            case 'bson':
+                assert($document instanceof Document);
+                $document->get('entities')->get('user_mentions')->get(0)->get('screen_name');
+                break;
+
+            case 'object':
+                assert(is_object($document));
```

Comment:
`Document` is also an object.

```suggestion
                assert($document instanceof stdClass);
```

Reply:
That is not true for the "Library default typemap" set, as the document will be returned as `BSONDocument` instance.